internal/overlord/servstate/handlers.go

```diff
@@ -162,11 +169,48 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 			}
 			releasePlan()
 		}
-		return fmt.Errorf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+
+		logs, err := getLastLogs(logBuffer)
+		if err != nil {
+			logger.Noticef("Cannot read service %q logs: %v", req.Name, err)
```

Comment:
Instead of logging locally, this should be bundled on the error message itself, as we're already raising an error, and this almost certainly is part of that error issue and should be observed and understood together.